exporters/memory/src/in_memory_metric_data.cc

```diff
@@ -31,7 +31,8 @@ void SimpleAggregateInMemoryMetricData::Add(std::unique_ptr<ResourceMetrics> res
       const auto &metric = m.instrument_descriptor.name_;
       for (const auto &pda : m.point_data_attr_)
       {
-        data_[{scope, metric}].insert({pda.attributes, pda.point_data});
```

Comment:
Instead of fixing it inline, could we create a separate macro like OPENTELEMETRY_LEGACY, and guard the C++ 11 fix in it? That way, we can find out and cleanup all the C++ 11 workarounds when it can be dropped in future.

Reply:
Or if introducing a new macro requires some extra work, perhaps simply add a comment here so say the C++ 11 workaround applied here, so we can locate all of them together later?

Reply:
I believe this was agreed in today's meeting to instead add a `TODO:` at places where are adding the C++11 workarounds - please correct me otherwise.

Reply:
@lalitb yes. As my second comment mentioned, they easy way would be adding a comment (like TODO) here, so we can find all the C++ 11 workarounds when we can drop the C++ 11 support (assume fully dropping C++ 11 support is still our long term goal).

Reply:
C++ permits a single implicit conversion of a function parameter type. It's more like a bug of GCC 4.8, and I'm not sure if other versions or compilers have similar issues.